src/sentry/api/endpoints/integrations/sentry_apps/components.py

```diff
@@ -54,13 +55,14 @@ def get(self, request: Request, organization) -> Response:
                     sentry_app_components.Preparer.run(
                         component=component, install=install, project=project
                     )
-                    components.append(component)
                 except APIError:
-                    continue
+                    errors.append(str(component.uuid))
+
+                components.append(component)
```

Comment:
Did you want to append the component if there's an APIError?

Reply:
Yeah, I need some way for the front end to read all the info about it but indicate there was an error and not to render it.